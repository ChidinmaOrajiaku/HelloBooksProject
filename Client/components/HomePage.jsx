import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import BodyNavigationBar from './BodyNavigationBar';
import Footer from './Footer';
import img1 from '../img/smiley1.jpg';
import img2 from '../img/smiley2.jpg';
import img3 from '../img/smiley3.jpg';
import img4 from '../img/sidney1.jpg';
import img5 from '../img/sideny2.jpeg';
import img6 from '../img/sidney3.jpg';
import img7 from '../img/sidney4.jpg';
import img8 from '../img/sidney5.jpg';
import img9 from '../img/sidney6.jpg';

export default () => (
  <div className="main">
    <div className="bodyNavigation"> <BodyNavigationBar /> </div>
    <div className="row">
      <div className="container first-section">
        <h1 className="welcome">
          Welcome to Hello Books
        </h1>
        <div>
          <Link to="/login"
            className="waves-effect waves-light btn col s2 offset-m3 "> Login
          </Link>
          <Link className="waves-effect waves-light btn col s2 offset-m1"
            to="/register"> Sign Up </Link>
        </div>
      </div>
      <div className="second-section">
        <div className="row">
          <div className="col s4" data-aos="zoom-in">
            <div>
              <h1>
                <img src={img1} data-aos="flip-left" />
                <br/>
            Read a book
              </h1>
              <span>
          Hi Reader! Looking for a book to read? You are in the right place.
          Just sign up with your awesome details and begin a wonderful book sail.
              </span>
            </div>
          </div>
          <div className="col s4" data-aos="zoom-in">
            <h1>
              <img src={img2} data-aos="flip-right" />
              <br/>
            Borrow a book
            </h1>
            <span>
          You can decide to borrow a book a save it for later.
          Yeah! It's your library, use it however you like.
          Just remember to return the borrowed books within the stipulated time.
            </span>
          </div>
          <div className="col s4" data-aos="zoom-in">
            <h1>
              <img src={img3} data-aos="flip-left"/>
              <br/>
            Suggest a book
            </h1>
            <span>
          Do you have any book you would love to read but is not available on our app?
          Or  have you read all available books and in need something new?
          Just suggest it to us and you'll get it in no time. Rememeber, your wish is our command.
            </span>
          </div>
        </div>
      </div>
      <div className="third-section">
        <div className="row container">
          <span className="col s12" data-aos="fade-up">Popular books</span>
          <div className="col s4" data-aos="fade-up">
            <img src={img4}/>
          </div>
          <div className="col s4" data-aos="fade-up">
            <img src={img5}/>
          </div>
          <div className="col s4" data-aos="fade-up">
            <img src={img6}/>
          </div>
          <div className="col s4" data-aos="fade-up">
            <img src={img7}/>
          </div>
          <div className="col s4" data-aos="fade-up">
            <img src={img8}/>
          </div>
          <div className="col s4" data-aos="fade-up">
            <img src={img9}/>
          </div>
        </div>
      </div>
    </div>
    <div> <Footer /></div>
  </div>
);

