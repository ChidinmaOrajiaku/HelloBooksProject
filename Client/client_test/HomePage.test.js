import React from 'react';
import $ from 'jquery';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import  HomePage  from '../components/HomePage.jsx';

configure({ adapter: new Adapter() });

describe('<HomePage />', () => {
  it('should render component without breaking', () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper).toBeDefined();
  });
  it('should match snapshot test', () => {
    const component = shallow(<HomePage />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});