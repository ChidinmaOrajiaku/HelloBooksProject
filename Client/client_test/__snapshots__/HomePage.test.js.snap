// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match snapshot test 1`] = `
<div
  className="main"
>
  <div
    className="bodyNavigation"
  >
     
    <Component />
     
  </div>
  <div
    className="row"
  >
    <div
      className="container first-section"
    >
      <h1
        className="welcome"
      >
        Welcome to Hello Books
      </h1>
      <div>
        <Link
          className="waves-effect waves-light btn col s2 offset-m3 "
          replace={false}
          to="/login"
        >
           Login
        </Link>
        <Link
          className="waves-effect waves-light btn col s2 offset-m1"
          replace={false}
          to="/register"
        >
           Sign Up 
        </Link>
      </div>
    </div>
    <div
      className="second-section"
    >
      <div
        className="row"
      >
        <div
          className="col s4"
          data-aos="zoom-in"
        >
          <div>
            <h1>
              <img
                data-aos="flip-left"
                src={Object {}}
              />
              <br />
              Read a book
            </h1>
            <span>
              Hi Reader! Looking for a book to read? You are in the right place. Just sign up with your awesome details and begin a wonderful book sail.
            </span>
          </div>
        </div>
        <div
          className="col s4"
          data-aos="zoom-in"
        >
          <h1>
            <img
              data-aos="flip-right"
              src={Object {}}
            />
            <br />
            Borrow a book
          </h1>
          <span>
            You can decide to borrow a book a save it for later. Yeah! It's your library, use it however you like. Just remember to return the borrowed books within the stipulated time.
          </span>
        </div>
        <div
          className="col s4"
          data-aos="zoom-in"
        >
          <h1>
            <img
              data-aos="flip-left"
              src={Object {}}
            />
            <br />
            Suggest a book
          </h1>
          <span>
            Do you have any book you would love to read but it's not available on our app? Or  have you read all available books and in need something new? Just suggest it to us and you'll get it in no time. Rememeber, your wish is our command.
          </span>
        </div>
      </div>
    </div>
    <div
      className="third-section"
    >
      <div
        className="row container"
      >
        <span
          className="col s12"
          data-aos="fade-up"
        >
          Popular books
        </span>
        <div
          className="col s4"
          data-aos="fade-up"
        >
          <img
            src={Object {}}
          />
        </div>
        <div
          className="col s4"
          data-aos="fade-up"
        >
          <img
            src={Object {}}
          />
        </div>
        <div
          className="col s4"
          data-aos="fade-up"
        >
          <img
            src={Object {}}
          />
        </div>
        <div
          className="col s4"
          data-aos="fade-up"
        >
          <img
            src={Object {}}
          />
        </div>
        <div
          className="col s4"
          data-aos="fade-up"
        >
          <img
            src={Object {}}
          />
        </div>
        <div
          className="col s4"
          data-aos="fade-up"
        >
          <img
            src={Object {}}
          />
        </div>
      </div>
    </div>
  </div>
  <div>
     
    <Component />
  </div>
</div>
`;
